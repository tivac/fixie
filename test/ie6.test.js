"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie6", () => {
    it("should transform :ie6(...) into an ie6 hack (single selector)", () =>
        simple(`
            :ie6(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie6(...) into an ie6 hack (multiple selectors)", () =>
        simple(`
            :ie6(.fooga .booga) {
                color: red;
            }
        `)
    );

    it("should transform :ie6(...) into an ie6 hack (multiple declarations)", () =>
        simple(`
            :ie6(.fooga) { color: red; background: blue; }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie6(&) {
                    color: blue;
                }
            }
        `)
    );
});
