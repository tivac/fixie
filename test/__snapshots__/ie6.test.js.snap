// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie6 should transform :ie6(...) into an ie6 hack (multiple declarations) 1`] = `".fooga { _color: red; _background: blue; }"`;

exports[`:ie6 should transform :ie6(...) into an ie6 hack (multiple selectors) 1`] = `
".fooga .booga {
    _color: red;
}"
`;

exports[`:ie6 should transform :ie6(...) into an ie6 hack (single selector) 1`] = `
".fooga {
    _color: red;
}"
`;

exports[`:ie6 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    .fooga {
        _color: blue;
    }"
`;
